src/app/pages/applications/catalog/catalog.component.ts

```diff
@@ -70,44 +72,54 @@ export class CatalogComponent implements OnInit {
     }
 
   ngOnInit(): void {
-    this.appService.getAllCatalogItems().subscribe(res => {
-      res.forEach(catalog => {
-        for (let i in catalog.trains.charts) {  
-          let item = catalog.trains.charts[i];
-          let versions = item.versions;
-          let latest, latestDetails;
-
-          let sorted_version_labels = Object.keys(versions);
-          sorted_version_labels.sort(this.utils.versionCompare);
-
-          latest = sorted_version_labels[0];
-          latestDetails = versions[latest];
-
-          let catalogItem = {
-            name: item.name,
-            catalog: {
-              id: catalog.id,
-              label: catalog.label,
-            },
-            icon_url: item.icon_url? item.icon_url : '/assets/images/ix-original.png',
-            latest_version: item.versions[latest].human_version,
-            info: latestDetails.app_readme,
-            schema: item.versions[latest].schema,
-          }
-          this.catalogApps.push(catalogItem);
-        }
-      });
-      this.filerApps();
-    })
-    
+    this.loadCatalogs();
     this.checkForConfiguredPool();
     this.refreshForms();
     this.refreshForm = this.modalService.refreshForm$.subscribe(() => {
       this.refreshForms();
     });
+  }
 
-    this.refreshTable = this.modalService.refreshTable$.subscribe(() => {
-      this.updateTab.emit({name: 'SwitchTab', value: '1'});
+  loadCatalogs() {
+    this.appService.getAllCatalogItems().subscribe(res => {
+      this.catalogNames = [];
+      this.catalogApps = [];
+      res.forEach(catalog => {
+        this.catalogNames.push(catalog.label);
+        catalog.preferred_trains.forEach(train => {
+          for (let i in catalog.trains[train]) {  
+            let item = catalog.trains.charts[i];
+            let versions = item.versions;
+            let latest, latestDetails;
```

Comment:
This is the thing breaking all trains not calling "charts".
Fix is relatively easy:

```suggestion
      res.forEach(catalog => {
        this.catalogNames.push(catalog.label);
        catalog.preferred_trains.forEach(train => {
          for (let i in catalog.trains[train]) {  
            let item = catalog.trains[train][i];
            let versions = item.versions;
            let latest, latestDetails;
```